check/file_extension.go

```diff
@@ -25,20 +25,16 @@ var ValidRegistryFileExtensions = []string{
 }
 
 func LegacyFileExtensionCheck(path string) error {
-	fileExtension := GetFileExtension(path)
-
-	if !IsValidLegacyFileExtension(fileExtension) {
-		return fmt.Errorf("invalid file extension (%s), valid values: %v", fileExtension, ValidLegacyFileExtensions)
+	if !FilePathEndsWithExtensionFrom(path, ValidLegacyFileExtensions) {
+		return fmt.Errorf("file does not end with a valid extension (%s), valid values: %v", GetFileExtension(path), ValidLegacyFileExtensions)
```

Comment:
`GetFileExtension` is still being used in the error message here, which means it could show the wrong value -- we should fully remove `GetFileExtension` (the function declaration and its unit testing) since its only used here and below.

```suggestion
		return fmt.Errorf("file does not end with a valid extension (%s), valid values: %v", path, ValidLegacyFileExtensions)
```